Automattic/pocket-casts-ios
# Reimagine Sharing: Export audio clips

| 📘 Part of: #1910  |
|:---:|

Adds exporting of audio only clips:

https://github.com/user-attachments/assets/0b02730b-b0e2-4388-b166-dac687c44a07

### Known Issues
* The current "clip.m4a" file name is a placeholder. File names cannot be dynamic with the `TransferRepresentation` APIs until iOS 17 😮‍💨 so I'll be changing the sharing code back to `NSItemProvider` in a separate PR.
* The title and description should change when swiping to the Audio style but I need to do a bit of refactoring to prevent a jump with the `TabView`.

## To test

> [!NOTE]
> Enable the `newSharing` feature flag to test this

* Play an episode
* Tap on the "Share" button in the action shelf
* Tap on "Clip" from the share modal
* Create a clip and swipe over the the audio symbol
* Tap "Clip"
* Tap the "More" share option and share or save to a file
* Verify that an `m4a` audio file is shared
* Verify that the `m4a` file is playable with correct audio

## Checklist
